### PR TITLE
[MOB-2711] Adyen Payment Details Endpoint

### PR DESCRIPTION
**Pull request by EdwardW2**
## JIRA
[MOB-2711](https://karhoo.atlassian.net/browse/MOB-2711)

## DESCRIPTION
Lots of currently unused models added. Some of this will need to be changed in the future but just adding it in for now as done with iOS. 

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [x] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [x] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: sdk/src/main/java/com/karhoo/sdk/api/model/adyen/PaymentsDetailsRequestPayload.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.karhoo.sdk.api.model.adyen
+
+import android.os.Parcelable
+import com.google.gson.annotations.SerializedName
+import kotlinx.android.parcel.Parcelize
+
+@Parcelize
+data class PaymentsDetailsRequestPayload(
```
Comment (jsm20) [inline sdk/src/main/java/com/karhoo/sdk/api/model/adyen/PaymentsDetailsRequestPayload.kt:8]: Are you using this anywhere?

Comment (EdwardW2) [reply sdk/src/main/java/com/karhoo/sdk/api/model/adyen/PaymentsDetailsRequestPayload.kt:8]: Not at the moment, Added it in from the iOS work on a duplicate of this ticket.

I can remove all of the unused models I have added in so far from iOS?

File: sdk/src/test/java/com/karhoo/sdk/api/service/payments/AdyenPaymentsDetailsInteractorTest.kt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        interactor = AdyenPaymentsDetailsInteractor(credentialsManager, apiTemplate, context)
+    }
+
+    /**
+     * Given:   A request is made to get Adyen payments
+     * When:    The call is not successful
+     * Then:    An InternalSDKError is returned
+     **/
+    @Test
+    fun `get Adyen payments call failure not made if request is null`() {
```
Comment (jsm20) [inline sdk/src/test/java/com/karhoo/sdk/api/service/payments/AdyenPaymentsDetailsInteractorTest.kt:35]: Minor comment but could you update so the comments and tests refer to Adyen payment details and not just Adyen payments

Comment (EdwardW2) [reply sdk/src/test/java/com/karhoo/sdk/api/service/payments/AdyenPaymentsDetailsInteractorTest.kt:35]: Ah yes. Good shout

**Review by jsm20 — DISMISSED** [review]

**Review by jsm20 — APPROVED** [review]

**Pull request merged**